ui-core/src/test/java/com/adyen/checkout/components/ui/SubmitHandlerTest.kt

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) 2023 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by oscars on 2/2/2023.
+ */
+
+package com.adyen.checkout.components.ui
+
+import androidx.lifecycle.SavedStateHandle
+import com.adyen.checkout.components.PaymentComponentState
+import com.adyen.checkout.components.model.payments.request.PaymentComponentData
+import com.adyen.checkout.components.ui.SubmitHandler.Companion.IS_INTERACTION_BLOCKED
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.coroutines.test.runCurrent
+import kotlinx.coroutines.test.runTest
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import java.util.UUID
+
+@OptIn(ExperimentalCoroutinesApi::class)
+internal class SubmitHandlerTest {
+
+    lateinit var submitHandler: SubmitHandler<PaymentComponentState<*>>
+
+    @Nested
+    @DisplayName("when initializing and ")
+    inner class InitializeTest {
+
+        @Test
+        fun `isInteractionBlocked is set, then UI state should be emitted`() = runTest {
+            val savedStateHandle = SavedStateHandle(mapOf(IS_INTERACTION_BLOCKED to true))
+            submitHandler = createSubmitHandler(savedStateHandle)
+
+            submitHandler.initialize(CoroutineScope(UnconfinedTestDispatcher()), flowOf())
+
+            runCurrent()
```

Comment:
Why do we need this?

Reply:
This makes sure all the events in the dispatcher are ran, so we can get the latest value of a flow

Reply:
Nevermind apparently it's not needed